packages/fiori/src/UploadCollection.js

```diff
@@ -0,0 +1,355 @@
+		document.body.removeEventListener("dragleave", this._dragleave);
+		document.body.removeEventListener("drop", this._ondrop);
+		this._globalHandlersAttached = false;
+	},
+
+	addUploadCollectionInstance(uploadCollection) {
+		this._uploadCollections.add(uploadCollection);
+		this._attachGlobalHandlers();
+	},
+
+	removeUploadCollectionInstance(uploadCollection) {
+		this.uploadCollections.delete(uploadCollection);
+
+		if (this.uploadCollections.size === 0) {
+			this._detachGlobalHandlers();
+		}
+	},
+};
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-upload-collection",
+	properties: /** @lends sap.ui.webcomponents.fiori.UploadCollection.prototype */ {
+		mode: {
+			type: ListMode,
+			defaultValue: ListMode.None,
+		},
+		noDataDescription: {
+			type: String,
+		},
+		noDataText: {
+			type: String,
+		},
+		noDnd: {
+			type: Boolean,
+		},
+		_dndOverlayMode: {
+			type: String,
+			defaultValue: DndOverlayMode.None,
+		},
+	},
+	managedSlots: true,
+	slots: /** @lends sap.ui.webcomponents.fiori.UploadCollection.prototype */ {
+		"default": {
+			propertyName: "items",
+			type: HTMLElement,
+		},
+		header: {
+			type: HTMLElement,
+		},
+	},
+	events: /** @lends sap.ui.webcomponents.fiori.UploadCollection.prototype */ {
+		fileDeleted: {
+			detail: {
+				item: { type: HTMLElement },
+			},
+		},
+		fileRenamed: {
+			detail: {
+				item: { type: HTMLElement },
+			},
+		},
+		/**
+		 * Fired when selection is changed by user interaction
+		 * in <code>SingleSelect</code> and <code>MultiSelect</code> modes.
+		 *
+		 * @event
+		 * @param {Array} selectedItems An array of the selected items.
+		 * @public
+		 */
+		selectionChange: {
+			detail: {
+				selectedItems: { type: Array },
+			},
+		},
+	},
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ *
+ * <h3>Usage</h3>
+ *
+ * For the <code>ui5-upload-collection</code>
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents-fiori/dist/UploadCollection.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.fiori.UploadCollection
+ * @extends UI5Element
+ * @tagname ui5-upload-collection
+ * @public
+ */
+class UploadCollection extends UI5Element {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return UploadCollectionCss;
+	}
+
+	static get template() {
+		return UploadCollectionTemplate;
+	}
+
+	static async onDefine() {
+		await Promise.all([
+			List.define(),
+			fetchI18nBundle("@ui5/webcomponents"),
+		]);
+	}
+
+	constructor() {
+		super();
+
+		this.i18nBundle = getI18nBundle("@ui5/webcomponents");
+		this._listeners = {
+			dragenter: this._ondragenter.bind(this),
+			dragleave: this._ondragleave.bind(this),
+			dragover: this._ondragover.bind(this),
+			drop: this._ondrop.bind(this),
+		};
+
+		this.addEventListener("_rename", this._onFileRenamed);
+	}
+
+	onEnterDOM() {
+		if (!this.noDnd) {
```

Comment:
Double negative logic is hard to read - please create a getter called needsDnd or similar

Reply:
I have replace it with if (this.noDnd) { return }
Hope it is more readable